akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -64,7 +66,65 @@ final case class HttpCookie(
   path:      Option[String]   = None,
   secure:    Boolean          = false,
   httpOnly:  Boolean          = false,
-  extension: Option[String]   = None) extends jm.headers.HttpCookie with ToStringRenderable {
+  extension: Option[String]   = None,
+  sameSite:  Option[SameSite] = None) extends jm.headers.HttpCookie with ToStringRenderable with Product with Serializable with Equals {
+
+  @deprecated("for binary compatibility", since = "10.2.0")
+  private[http] def this(
+    name:      String,
+    value:     String,
+    expires:   Option[DateTime],
+    maxAge:    Option[Long],
+    domain:    Option[String],
+    path:      Option[String],
+    secure:    Boolean,
+    httpOnly:  Boolean,
+    extension: Option[String]) = this(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, None)
+
+  private[http] def copy(sameSite: Option[SameSite]) = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, sameSite)
+
+  def copy(
```

Comment:
Going forward, we would probably rather provide `withXYZ` methods instead of a copy method that is hard to evolve. Maybe we should do this right now and deprecate the copy as well?

Reply:
Do we have a decision on that one? @raboof, wdyt?

Reply:
In #2938 I kept one but made it `private[package]` - that way it's nice for implementing those `withXyz` methods but we don't need to keep it binary compatible going forward.

Reply:
though I don't mind having it public much either, even if it leaves behind some cruft each time we add a field

Reply:
> In #2938 I kept one but made it `private[package]` - that way it's nice for implementing those `withXyz` methods but we don't need to keep it binary compatible going forward.

Let's go with this way.